src/greenlight_alby_client.rs

```diff
@@ -0,0 +1,217 @@
+        GreenlightInvoiceResponse {
+            bolt11: invoice.bolt11,
+        }
+    }
+}
+
+pub struct GreenlightAlbyClient {
+    // signer: gl_client::signer::Signer,
+    scheduler: gl_client::scheduler::Scheduler,
+    tls: TlsConfig,
+}
+
+pub async fn recover(mnemonic: String) -> Result<GreenlightCredentials> {
+    let mnemonic = Mnemonic::from_str(&mnemonic)
+        .context("failed to parse mnemonic")
+        .map_err(SdkError::invalid_arg)?;
+
+    let secret = mnemonic.to_seed("")[0..32].to_vec(); // Only need the first 32 bytes
+
+    let tls = TlsConfig::new()
+        .context("failed to create TLS config")
+        .map_err(SdkError::greenlight_api)?;
+
+    let signer = Signer::new(secret, Network::Bitcoin, tls)
+        .context("failed to create signer")
+        .map_err(SdkError::greenlight_api)?;
+
+    let scheduler = Scheduler::new(signer.node_id(), Network::Bitcoin)
+        .await
+        .context("failed to create scheduler")
+        .map_err(SdkError::greenlight_api)?;
+
+    Ok(scheduler
+        .recover(&signer)
+        .await
+        .context("failed to recover credentials")
+        .map_err(SdkError::greenlight_api)?
+        .into())
+}
+
+pub async fn new_greenlight_alby_client(
+    mnemonic: String,
+    credentials: GreenlightCredentials,
+) -> Result<Arc<GreenlightAlbyClient>> {
+    let tls = TlsConfig::new()
+        .context("failed to create TLS config")
+        .map_err(SdkError::greenlight_api)?
+        .identity(
+            credentials.device_cert.into_bytes(),
+            credentials.device_key.into_bytes(),
+        );
+
+    let mnemonic = Mnemonic::from_str(&mnemonic)
+        .context("failed to parse mnemonic")
+        .map_err(SdkError::invalid_arg)?;
+
+    let secret = mnemonic.to_seed("")[0..32].to_vec(); // Only need the first 32 bytes
+
+    let signer = Signer::new(secret, Network::Bitcoin, tls.clone())
+        .context("failed to create signer")
+        .map_err(SdkError::greenlight_api)?;
+    let scheduler = Scheduler::new(signer.node_id(), Network::Bitcoin)
+        .await
+        .context("failed to create scheduler")
+        .map_err(SdkError::greenlight_api)?;
+
+    Ok(Arc::new(GreenlightAlbyClient {
+        tls,
+        scheduler,
+        // signer,
+    }))
+}
+
+impl GreenlightAlbyClient {
+    async fn get_node(&self) -> Result<gl_client::node::ClnClient> {
+        // wakes up the node
+        self.scheduler
+            .schedule(self.tls.clone())
+            .await
+            .context("failed to schedule node")
+            .map_err(SdkError::greenlight_api)
+    }
+
+    pub async fn get_info(&self) -> Result<GreenlightNodeInfo> {
+        let mut node = self.get_node().await?;
+
+        node.getinfo(cln::GetinfoRequest::default())
+            .await
+            .context("failed to get node info")
+            .map_err(SdkError::greenlight_api)
+            .map(|r| r.into_inner().into())
+    }
+
+    pub async fn make_invoice(
```

Comment:
Can we make this return a pointer? so in go the value would be nil if there is an error